src/emotion-tracking/emotion-tracking.service.ts

```diff
@@ -0,0 +1,130 @@
+            emotionalRecord.date = currentDate;
+            emotionalRecord.emotionalLevel = Emotion[emotionLevel];
+            const user = await this.userRepository.findOne({uid});
+            emotionalRecord.user = user;
+            await this.emotionRecordRepository.save(emotionalRecord);  
+            return {message: "Emotion record successfully created"};
+        } 
+        throw new HttpException('Emotion record of this elderly has already been made today', HttpStatus.BAD_REQUEST);
+    }
+
+    async getEmotionalRecord(caretakerId: number, elderlyId: number, limit: number, offset:number=0): Promise<EmotionHistory>{
+        await this.userService.checkRelationship(elderlyId, caretakerId);
+        let emotionalRecord;
+        let numRecords;
+        emotionalRecord = await this.emotionRecordRepository
+            .createQueryBuilder("emotionRecord")
+            .where("emotionRecord.uid = :uid", {uid: elderlyId})
+            .orderBy("emotionRecord.date", "DESC")
+            .getMany();
+        numRecords = emotionalRecord.length;
+    
+        //check if the limit is specified. default: return all
+        if(limit){
+            emotionalRecord = await this.emotionRecordRepository
+                .createQueryBuilder("emotionRecord")
+                .where("emotionRecord.uid = :uid", {uid: elderlyId})
+                .orderBy("emotionRecord.date", "DESC")
+                .limit(limit)
+                .offset(offset)
+                .getMany();
+        }
+        emotionalRecord.forEach(record => {
+            record.emotionalLevel = Emotion[record.emotionalLevel]
+        })
+        const emotionalHistory = {
+            count: numRecords,
+            records: emotionalRecord
+        }
+        return emotionalHistory;
+    }
+
+    async getEmotionTrackingStatus(caretakerId: number, elderlyId: number): Promise<EmotionTrackingStatusDto>{
+        await this.userService.checkRelationship(elderlyId, caretakerId);
+        const elderly = await this.userRepository.findOne({uid: elderlyId},{
+            relations: ["modules"]
+        })
+        const moduleFive = elderly.modules.filter(module=> module.moduleid === 5)[0];
+        let emotionTrackingStatus = new EmotionTrackingStatusDto();
+        if(moduleFive){
+            emotionTrackingStatus['isEnabled'] = true;
+        } else{
+            emotionTrackingStatus['isEnabled'] = false;
+        }
+        return emotionTrackingStatus;
+    }
+
+    async addEmotionalTrackingModuleToElderly(caretakerId: number, elderlyId: number): Promise<{statusCode: number, message: string}>{
+        await this.userService.checkRelationship(elderlyId, caretakerId);
+        const elderly = await this.userRepository.findOne({uid: elderlyId},{
+            relations: ["modules"]
+        })
+        const moduleFive = elderly.modules.filter(module=> module.moduleid === 5)[0];  //5 is the moduleid of Emotion Tracking module
```

Comment:
this is nothing serious but use `.find` so it doesn't have to scan the whole array since you only need the first one anyway.